### DISCUSSION TITLE
Recommended way to use additional textlint rules in a github action

### DISCUSSION THREAD

**Discussion by kukimik** [super-linter/super-linter#5222] [Q&A] [answered]
Hi, and thanks for the work on super-linter!

I want to use super-linter as a github action. In particular I want to use the `textlint` linter with some specific set of rules (`textlint` installs no rules by default; they need to be installed using `npm install`; see https://github.com/textlint/textlint?tab=readme-ov-file#usage). In `super-linter` only one `textlint` rule is included (the `textlint-rule-terminology` rule): https://github.com/super-linter/super-linter/blob/5c67776f9d9d3065efda30d47aef21b332fa2173/dependencies/package.json#L54

What is the recommended way to set up the github action to use additional rules, e.g. from here https://github.com/textlint-rule?

**Comment by ferrarimarco** [comment, marked as answer]
Hi!

This may be the way forward to get the textlint rules you're interested in:

- If the rules work with `.txt` and `.md` files, we can install it. In this case, you can also build your container image using `super-linter` as a base and install the rules you're missing.
- If the rules work with additional file extensions and file types, installing them is not going to be enough because we also need to add the logic to make super-linter aware of these new file types. It's not hard to do, but it's necessary.

For both cases, I would start by opening a new feature request in this repository listing the rules you'd like to include :)
